CodeHawk/README.md

```diff
@@ -8,166 +8,39 @@ Building codehawk requires the following applications and libraries:
-setup above to get the dependencies.
+opam init --bare
 
-### Generic Unix
-
-Other dependencies: `pkg-config`, `m4`
-
-```
-sudo apt update -y
-sudo apt install curl -y
-curl -sSL https://get.haskellstack.org/ | sh
-sh <(curl -sL https://raw.githubusercontent.com/ocaml/opam/master/shell/install.sh)
-stack install shake
 git clone https://github.com/static-analysis-engineering/codehawk.git
 cd codehawk/CodeHawk
-stack runghc Shakefile.hs
-```
-
-### Arch Linux
+opam switch create . 5.2.0
+eval $(opam env)
+opam install dune ocamlfind zarith camlzip extlib goblint-cil.2.0.6
 
-```
-sudo pacman -Syu opam haskell-shake lablgtk2
-git clone https://github.com/static-analysis-engineering/codehawk.git
-cd codehawk/CodeHawk
-shake
-```
-
-### Fedora
-
-```
-sudo yum install opam shake ghc-compiler ghc-shake-devel diffutils zlib-devel ocaml-lablgtk-devel -y
-git clone https://github.com/static-analysis-engineering/codehawk.git
-cd codehawk/CodeHawk
-shake
+dune build @install
 ```
 
-### Homebrew
+The Makefiles in the repository are to help CodeHawk's developers
+debug circular module dependencies, they are not intended for users.
 
-```
-brew install opam
-brew install cabal-install
-brew install lablgtk
-cabal update
-cabal install --lib shake
-cabal install --lib unordered-containers
-git clone https://github.com/static-analysis-engineering/codehawk.git
-cd codehawk/CodeHawk
-~/.cabal/bin/shake
-```
-
-### MacPorts
-
-```
-sudo port install opam
-sudo port install hs-cabal-install
-sudo port install lablgtk2
-cabal update
-cabal install --lib shake
-cabal install --lib unordered-containers
-git clone https://github.com/static-analysis-engineering/codehawk.git
-cd codehawk/CodeHawk
-~/.cabal/bin/shake
-```
-
-### Ubuntu 19.04 or later, Debian Buster or later
-
-```
-sudo apt update -y
-sudo apt install cabal-install pkg-config m4 zlib1g-dev opam liblablgtk2-ocaml-dev liblablgtksourceview2-ocaml-dev -y
-cabal update
-cabal install shake
-git clone https://github.com/static-analysis-engineering/codehawk.git
-cd codehawk/CodeHawk
-~/.cabal/bin/shake
-```
-
-### Ubuntu 16.04 or later
-
-```
-sudo apt update -y
-sudo apt install software-properties-common cabal-install pkg-config m4 zlib1g-dev liblablgtk2-ocaml-dev liblablgtksourceview2-ocaml-dev -y
-sudo add-apt-repository ppa:avsm/ppa -y
-sudo apt update -y
-sudo apt install opam
-cabal update
-cabal install shake
-git clone https://github.com/static-analysis-engineering/codehawk.git
-cd codehawk/CodeHawk
-~/.cabal/bin/shake
-```
-
-### Manually configured opam
-
-Before the final "shake" command in one of the above instructions:
-
-```
-opam init --bare --no-setup --disable-sandboxing
-opam switch create codehawk-4.12.1 4.09.0 --no-switch
-eval $(opam env --switch=codehawk-4.12.1 --set-switch)
-opam install ocamlfind zarith camlzip extlib lablgtk lablgtk-extras
-/path/to/shake
-```
+Dependencies for other OS flavors:
```

Comment:
Rather than organizing the page as

- Dependencies (assuming Debian/Ubuntu)
- Install instructions
- Dependencies (other distros)

can the page use [collapsed sections](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections) to group all the operating system / distro instructions together without taking up too much space? That doesn't favor any operating system or distribution in particular.